chain/chunks/src/lib.rs

```diff
@@ -722,6 +722,12 @@ impl ShardsManager {
             if !cares_about_shard {
                 self.encoded_chunks.remove_from_cache_if_outside_horizon(&chunk_hash);
                 self.requested_partial_encoded_chunks.remove(&chunk_hash);
+                info!(
```

Comment:
We *always* want to trace it (I think I even emphasized it several times :)), not only when we don't care about shard, because this is the condition under which a chunk is included in the block. Whether or not we follow the shard, we include the chunk once we have all parts and receipts, we do not wait until we can reconstruct it.

Reply:
See my comment below, i thought i trace both case?

Reply:
i'm wrong, actually i only traced all the path it could return `HaveAllPartsAndReceipts`, care shard or not care, but seems there's a path it returns last `NeedMoreOnePartsOrReceipts` if need have_all_parts && have_all_receipts && cares_about_shard && !can_reconstruct